integrations/access/datadog/bot.go

```diff
@@ -153,6 +158,43 @@ func (b Bot) FetchRecipient(ctx context.Context, name string) (*common.Recipient
 	}, nil
 }
 
+// FetchOncallUsers fetches on-call users filtered by the provided annotations.
+func (b Bot) FetchOncallUsers(ctx context.Context, req types.AccessRequest) ([]string, error) {
+	log := logger.Get(ctx)
+
+	// Fetch all on-call teams
+	oncallTeams, err := b.datadog.GetOncallTeams(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	teamNames, err := common.GetNamesFromAnnotations(req, AutoApprovalsAnnotation)
+	if err != nil {
+		log.Debug("Automatic approvals annotation is empty or unspecified.")
+		return nil, nil
+	}
```

Comment:
Follow up on [the `teleportUser` discussion](https://github.com/gravitational/teleport/pull/47475/files/d4ce6a432625e29a48af6f31cdaf91c6091e5141#r1797467035): it seems this code will run on every request. In this case can we try to fail as fast as possible?

Here we could save many datadog calls by checking the annotations before looking up the on-call teams and returning early if auto-approval is not configured.

Reply:
Good call